modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/IngestUserAgentPlugin.java

```diff
@@ -67,7 +70,8 @@
                 for (Path path : iterable) {
                     String parserName = path.getFileName().toString();
                     try (InputStream regexStream = Files.newInputStream(path, StandardOpenOption.READ)) {
-                        userAgentParsers.put(parserName, new UserAgentParser(parserName, regexStream, cache));
+                        InputStream deviceTypeRegexStreamC = IngestUserAgentPlugin.class.getResourceAsStream("/device_type_regexes.yml");
```

Comment:
Move the declaration of `deviceTypeRegexStreamC` into the `try` portion of the try-with-resources block.